eglib/src/gmisc-win32.c

```diff
@@ -116,20 +117,32 @@ g_path_is_absolute (const char *filename)
 const gchar *
 g_get_home_dir (void)
 {
-	/* FIXME */
-	const gchar *drive = g_getenv ("HOMEDRIVE");
-	const gchar *path = g_getenv ("HOMEPATH");
 	gchar *home_dir = NULL;
-	
-	if (drive && path) {
-		home_dir = g_malloc(strlen(drive) + strlen(path) +1);
-		if (home_dir) {
-			sprintf(home_dir, "%s%s", drive, path);
-		}
+
+	PWSTR profile_path = NULL;
+	HRESULT hr = SHGetKnownFolderPath (&FOLDERID_Profile, KF_FLAG_DEFAULT, NULL, &profile_path);
```

Comment:
Thanks for taking a look! You are right, there seems to be a bug in the `configure.ac` in the root. It never passes on any defines at all to `eglib/configure`, like it does when invoking `libgc/configure`. Mingw does have the required header file and knows about `SHGetKnownFolderPath()` but only when `_WIN32_WINNT` is set to at least Vista. I'll see if I can sort that out too...